### PR TITLE
Fix EC2Hook get_instance for client_type api

### PR DESCRIPTION
**Pull request by hussein-awala**
The method `get_instance` should return a single instance and not a list.

### CODE REVIEW

File: airflow/providers/amazon/aws/hooks/ec2.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def get_instance(self, instance_id: str, filters: list | None = None):
         :return: Instance object
         """
         if self._api_type == "client_type":
-            return self.get_instances(filters=filters, instance_ids=[instance_id])
+            return self.get_instances(filters=filters, instance_ids=[instance_id])[0]
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/hooks/ec2.py:88]: Is not it breaking change? Or we consider it as bug fix?

Comment (hussein-awala) [reply airflow/providers/amazon/aws/hooks/ec2.py:88]: For me, it's a bug fix. We can check with @eladkal.

Comment (pankajkoti) [reply airflow/providers/amazon/aws/hooks/ec2.py:88]: The method seems to return a single instance with the default api_type="resource_type". Similarly, this method should return a single instance for other api_types, so I would treat it as a bug fix. Also, it's named as get_instance so the expectation would be a single instance to be returned.

**Review by pankajkoti — APPROVED** [review]

**Review by vincbeck — APPROVED** [review]

File: airflow/providers/amazon/aws/hooks/ec2.py (hunk 1 above)
Comment (eladkal) [reply airflow/providers/amazon/aws/hooks/ec2.py:88]: I would classify this as bug fix

**Review by eladkal — APPROVED** [review]

**Comment by hussein-awala** [comment]
Thanks to all of you!

**Pull request merged**